src/compiler/config/load-config.ts

```diff
@@ -44,6 +44,8 @@ export const loadConfig = async (init: LoadConfigInit = {}): Promise<LoadConfigR
     },
   };
 
+  const unknownConfig: UnvalidatedConfig = {};
```

Comment:
This change decouples the validation of a config from the `results` object that's declared at the top of this function. That allows us to create an `UnvalidatedConfig` here, adding entities to it as needed. Then we can pass the `UnvalidatedConfig` to the validator, and get a strict/internal one back.